lib/spack/spack/compiler.py

```diff
@@ -298,6 +293,13 @@ def check(exe):
             if value is not None:
                 self.flags[flag] = tokenize_flags(value)
 
+    def verify_executables(self):
```

Comment:
(minor) I liked the more-generic version that verified a collection of provided paths but this is alright (in that case the function could be placed in a more-generic module like `llnl.util.filesystem). 

I think a docstring would be useful.

Reply:
A doc-string is a good idea.

I was torn on the generic version. The version attached to a `Compiler` object allows us to write more useful error messages, which seemed compelling enough to me. But there's definitely something to be said for the generic version, do you think I should go back to it?

Reply:
No if you have any reason to stick with the current implementation don't change it: not a big deal.

Reply:
Ok yeah I'll leave it. I think including the spec of the compiler in the error message will make it much easier for users to fix when they run into this.